internal/host/validator.go

```diff
@@ -691,103 +605,41 @@ func getRealHostname(host *models.Host, inventory *models.Inventory) string {
-	for _, h := range c.cluster.Hosts {
-		if h.ID.String() != c.host.ID.String() && h.Inventory != "" {
-			otherInventory, err := c.inventoryCache.GetOrUnmarshal(h)
-			if err != nil {
-				v.log.WithError(err).Warnf("Illegal inventory for host %s", h.ID.String())
-				// It is not our hostname
-				continue
-			}
-			if realHostname == getRealHostname(h, otherInventory) {
-				return ValidationFailure
-			}
-		}
-	}
-	return ValidationSuccess
-}
-
-func (v *validator) printHostnameUnique(c *validationContext, status ValidationStatus) string {
-	switch status {
-	case ValidationSuccess:
-		return fmt.Sprintf("Hostname %s is unique in cluster", getRealHostname(c.host, c.inventory))
-	case ValidationFailure:
-		return fmt.Sprintf("Hostname %s is not unique in cluster", getRealHostname(c.host, c.inventory))
-	case ValidationPending:
-		return "Missing inventory"
-	default:
-		return fmt.Sprintf("Unexpected status %s", status)
-	}
-}
-
-func (v *validator) isHostnameValid(c *validationContext) ValidationStatus {
+func (v *validator) isHostnameValid(c *validationContext) (ValidationStatus, string) {
 	if c.inventory == nil {
-		return ValidationPending
+		return ValidationFailure, "Missing inventory"
 	}
 
 	if err := hostutil.ValidateHostname(getRealHostname(c.host, c.inventory)); err != nil {
-		return ValidationFailure
-	}
-
-	return ValidationSuccess
-}
-
-func (v *validator) printHostnameValid(c *validationContext, status ValidationStatus) string {
-	switch status {
-	case ValidationSuccess:
-		return fmt.Sprintf("Hostname %s is allowed", getRealHostname(c.host, c.inventory))
-	case ValidationFailure:
 		if funk.ContainsString(hostutil.ForbiddenHostnames, getRealHostname(c.host, c.inventory)) {
-			return fmt.Sprintf("The host name %s is forbidden", getRealHostname(c.host, c.inventory))
+			return ValidationFailure, fmt.Sprintf("The host name %s is forbidden", getRealHostname(c.host, c.inventory))
 		}
-		return fmt.Sprintf("Hostname %s is forbidden, hostname should match pattern %s", getRealHostname(c.host, c.inventory), hostutil.HostnamePattern)
-	case ValidationPending:
-		return "Missing inventory"
-	default:
-		return fmt.Sprintf("Unexpected status %s", status)
+		return ValidationFailure, fmt.Sprintf("Hostname %s is forbidden, hostname should match pattern %s", getRealHostname(c.host, c.inventory), hostutil.HostnamePattern)
 	}
+	return ValidationSuccess, fmt.Sprintf("Hostname %s is allowed", getRealHostname(c.host, c.inventory))
 }
 
-func (v *validator) isIgnitionDownloadable(c *validationContext) ValidationStatus {
+func (v *validator) isIgnitionDownloadable(c *validationContext) (ValidationStatus, string) {
 	if c.infraEnv != nil {
-		return ValidationSuccessSuppressOutput
+		return ValidationSuccessSuppressOutput, ""
 	}
-	if !hostutil.IsDay2Host(c.host) {
-		return ValidationSuccessSuppressOutput
+	if !hostutil.IsDay2Host(c.host) || swag.BoolValue(c.cluster.UserManagedNetworking) {
```

Comment:
This user managed network case used to be `ValidationSuccess`, not `ValidationSuccessSuppressOutput`

Looks like you actually have it handled below so maybe just remove that condition from here?

Reply:
Any ideas about this @paul-maidment ?